public/sw.js

```diff
@@ -0,0 +1,59 @@
+var CACHE_NAME = 'weather-10kb-cache-v1';
+var urlsToCache = [
+  '/'
+];
+
+self.addEventListener('install', function(event) {
+  event.waitUntil(
+    caches.open(CACHE_NAME)
+      .then(function(cache) {
+        console.log('Opened cache');
+        return cache.addAll(urlsToCache);
+      })
+  );
+});
+
+self.addEventListener("fetch", function(event) {
+  console.log('WORKER: fetch event in progress.');
+
+  if (event.request.method !== 'GET') {
+    console.log('WORKER: fetch event ignored.', event.request.method, event.request.url);
+    return;
+  }
+  event.respondWith(
+    caches
+      .match(event.request)
+      .then(function(cached) {
+        var networked = fetch(event.request)
+          .then(fetchedFromNetwork, unableToResolve)
+          .catch(unableToResolve);
+        console.log('WORKER: fetch event', cached ? '(cached)' : '(network)', event.request.url);
+        return cached || networked;
+
+        function fetchedFromNetwork(response) {
+          var cacheCopy = response.clone();
+          console.log('WORKER: fetch response from network.', event.request.url);
+          caches
+            .open(version + 'pages')
+            .then(function add(cache) {
+              cache.put(event.request, cacheCopy);
+            })
+            .then(function() {
+              console.log('WORKER: fetch response stored in cache.', event.request.url);
+            });
+          return response;
+        }
+
+        function unableToResolve () {
+          console.log('WORKER: fetch request failed in both cache and network.');
+          return new Response('<h1>Service Unavailable</h1>', {
```

Comment:
Future TODO: Return a clear statement as to what the error is, and how to fix it.